src/vis/bar/utils.ts

```diff
@@ -142,14 +142,14 @@ function setPlotsWithGroupsAndMultiples(columns, config, plots, scales, plotCoun
 
             plots.push({
                 data: {
-                    x: vertFlag ? [...new Set(catCurr.values.map((v) => v.val))] : groupedLength,
-                    y: !vertFlag ? [...new Set(catCurr.values.map((v) => v.val))] : groupedLength,
+                    x: vertFlag ? [...new Set(catCurr.resolvedValues.map((v) => v.val))] : groupedLength,
+                    y: !vertFlag ? [...new Set(catCurr.resolvedValues.map((v) => v.val))] : groupedLength,
                     orientation: vertFlag ? 'v' : 'h',
                     xaxis: plotCounter === 1 ? 'x' : 'x' + plotCounter,
                     yaxis: plotCounter === 1 ? 'y' : 'y' + plotCounter,
                     showlegend: plotCounter === 1 ? true : false,
                     type: 'bar',
-                    name: uniqueGroup,
+                    name: uniqueGroup as string,
```

Comment:
I assume Typescript infers `uniqueGroup` as `number|string`. I suggest to do the typecast as early as possible so that the downstream usage is easier and you do not need to typecast multiple times.

Reply:
correct, and agreed, made the change that the next comment suggests